### ISSUE TITLE
Broken build due to load tag with Rouge syntax highlighter

### ISSUE THREAD

**Issue by raybesiga** [jekyll/jekyll#4569]
Still experiencing this bug. I am using Jekyll 3.1.1 on Mac OS X 10.11.2 and I am using Rouge for syntax highlighting in a .markdown post file.

I use regular markdown for text and headers and the `highlight` command for special commands. I have encountered an issue while using `{% raw %}` and {% endraw %} to wrap code that I wanted Liquid to leave as is, as shown below:

```
Before we can show data in the `index.html` file, we need to get the context data within the views file. Let is make those changes to our `yokogao/views.py` file.

{% highlight django %}
import dateutil.parser
from django.shortcuts import render
import requests

def index(request):
    yoko = requests.get('https://api.github.com/users/raybesiga')
    context = yoko.json()

    # context has a created_at key with a string value
    # parse the string value into a python date object
    context['created_at'] = dateutil.parser.parse(context['created_at'])
    return render(request, 'yokogao/index.html', context)
{% endhighlight %}

Then edit the `index.html` file

{% raw %}
{% extends "base.html" %}

{% block title %} Github Yokogao {% endblock %}

{% block content %}
<div class="columns centered">
    <div class="one-third column centered">
        <h3>Basic Django app with the Github API</h3>
    </div>
</div>
<div class="columns">
  <div class="one-third column centered">
    <img class="avatar" src="{{ avatar_url }}" width="300" height="300" style="border-radius: 50%;">
  </div>
  <div class="one-third column centered">
    <p><br></p>
  </div>
  <div class="one-third column centered">
        <nav class="menu">
            <a class="menu-item" href="#">
            <span class="octicon octicon-person"></span>
            {{ name }}
          </a>
          <a class="menu-item" href="#">
            <span class="octicon octicon-location"></span>
            {{ location }}
          </a>
          <a class="menu-item" href="#">
            <span class="octicon octicon-list-ordered"></span>
            <span class="counter">{{ public_repos }}</span>
            Public Repos
          </a>
          <a class="menu-item" href="#">
            <span class="octicon octicon-bookmark" ></span>
            <span class="counter">{{ followers }}</span>
            Followers
          </a>
          <a class="menu-item" href="#">
            <span class="octicon octicon-eye"></span>
            <span class="counter">{{ following }}</span>
            Following
          </a>
          <a class="menu-item" href="#">
            <span class="octicon octicon-calendar"></span>
            Member for {{ created_at|timesince }}
          </a>
          <a class="menu-item " href="{{ blog }}" target="_blank">
            <span class="octicon octicon-globe"></span>
            Website: {{ blog }}
          </a>
        </nav>
    </div>
</div>
<div clas="columns">
    <div class="one-third column centered">
        <h4>Built with <span class="octicon octicon-heart" ></span> using the Github API and Primer</h4>
    </div>
</div>
{% endblock %}
{% endraw %}

Now if you run your server, you should have the browser showing your yokogao on the index page.
```

However, when I run `jekyll serve`, I get the following error:

```
test-site git:(master) ✗ jekyll serve --profile
Configuration file: /Users/raybesiga/Projects/test-site/_config.yml
            Source: /Users/raybesiga/Projects/test-site
       Destination: /Users/raybesiga/Projects/test-site/_site
 Incremental build: disabled. Enable with --incremental
      Generating...
  Liquid Exception: Liquid syntax error: Unknown tag 'load' in /Users/raybesiga/Projects/test-site/_posts/2016-02-23-basic-django-app-github-api.markdown
jekyll 3.1.1 | Error:  Liquid syntax error: Unknown tag 'load'
```
Will appreciate any help and support. Cheers.

**Comment by pathawks** [comment]
This is a duplicate of #4567

**Comment by TWiStErRob** [comment]
`{% raw %}` only suppresses Jekyll's Liquid processing, so this is what happens:
 1. File read from disk
 1. Jekyll magic with frontmatter and layouts
 1. Contents passed to Liquid  
`{% raw %}` and `{% highlight %}` blocks are interpreted and contents passed to the next level
 1. Now you have a Markdown file with some Django and HTML mixed in  
Kramdown (or whatever) interprets your Markdown mixed HTML  
Here any further {% ... %} is ignored, it's just plain text in Markdown
 1. Browser reads generated HTML tags and renders page (again, unaware of Django).

Based on the contents of your page I think you want to do this:
```django
Mine looks something like this:
{% highlight django %}{% raw %}
Django code with <html>, {% load %} and {% stylesheet %}
{% endraw %}{% endhighlight %}
```
*(finally I found a highlighter (django) that can highlight Liquid code on GitHub)*
or if you like GFM:
```markdown
Mine looks something like this:
```django{% raw %}
Django code with <html>{% load %} and {% stylesheet %}</html>
```{% endraw %}
```

Notice that the `{% highlight %}` and `{% raw %]` blocks are nested properly, because they're both processed by Liquid. Compare that to ```` ``` ```` and `{% raw %}` where first raw is processed and then the Markdown parser only gets a well-formed fenced block. This style helps with unsuspecting code editors which are not aware of all these different languages and they recognize the fenced block properly (at least what I tried).

Consider yourself lucky that you don't need to write about Liquid, outputting `{% raw %}` in a highlighted codeblock is a nightmare: http://blog.slaks.net/2013-06-10/jekyll-endraw-in-code/

**Comment by raybesiga** [comment]
@TWiStErRob Thanks for looking into it. I found a way around a day later by doing it this way:

```
{% highlight jinja %}
{% raw %}{% load pipeline %}{% endraw %}
<!DOCTYPE html>  
<html lang="en">  
  <head>
    <meta charset="utf-8">
    <title>{% raw %}{% block title %}Basic Django app with the Github API{% endblock %}{% endraw %}</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="description" content="Building with the Github API">
    <meta name="author" content="Ray Besiga">
    {% raw %}{% stylesheet 'yokogao' %}{% endraw %}
  </head>
  <body>
    <div id="layout" class="pure-g">
      <div class="sidebar pure-u-1 pure-u-md-1-4">
        <div class="header">
            <h1 class="brand-title">Basic Github App</h1>
            <h2 class="brand-tagline">Using Django, Requests, Bower and Pipeline</h2>
            <nav class="nav">
              <h3>Tools leveraged</h3>
                <ul class="nav-list">
                    <li class="nav-item">Github API</li>,
                    <li class="nav-item">Django</li>,
                    <li class="nav-item">Requests</li>,
                    <li class="nav-item">Bower</li>,
                    <li class="nav-item">Pipeline</li>,
                    <li class="nav-item">Primer CSS</li>,
                    <li class="nav-item">Pure Grid CSS</li>
                </ul>
            </nav>
        </div>
    </div>
    <div class="content pure-u-1 pure-u-md-3-4">
          <div>
          {% raw %}{% block content %}{% endblock %}{% endraw %}
          </div>
        </div>
      </div>
    </div>
    {% raw %}{% block extra_scripts %}{% endblock %}{% endraw %}
  </body>
</html>  
{% endhighlight %}
```
Clearly not as elegant as your solution. Let me also check out the Slaks blog. And promptly check out your Github :)

**Comment by jekyllbot** [comment]
This issue has been automatically marked as stale because it has not been commented on for at least
three months.

The resources of the Jekyll team are limited, and so we are asking for your help.

If you can still reproduce this error on the <code>3.1-stable</code> or <code>master</code> branch,
please reply with all of the information you have about it in order to keep the issue open.

If this is a feature request, please consider building it first as a plugin. Jekyll 3 introduced
[hooks](http://jekyllrb.com/docs/plugins/#hooks) which provide convenient access points throughout
the Jekyll build pipeline whereby most needs can be fulfilled. If this is something that cannot be
built as a plugin, then please provide more information about why in order to keep this issue open.

Thank you for all your contributions.

**Comment by TWiStErRob** [comment]
This could be closed as dupe of #4567, @raybesiga stated that he found `{%raw%}` to do the trick.

**Comment by parkr** [comment]
Thanks, @TWiStErRob!
